src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,44 @@
 
     @Override
     public User register(User user) {
+        if (userValidation(user)) {
+            return storageDao.add(user);
+        }
         return null;
     }
+
+    private boolean userValidation(User user) {
+        return userNullCheck(user)
+                && userValidateLoginExist(user.getLogin())
+                && userValidateLoginLength(user.getLogin())
+                && userValidatePasswordLength(user.getPassword())
+                && userValidateAge(user.getAge());
+    }
+
+    private boolean userNullCheck(User user) {
+        return user != null
+                && user.getLogin() != null
+                && user.getPassword() != null
+                && user.getAge() != null;
+    }
+
+    private boolean userValidateLoginExist(String userLogin) {
+        return storageDao.get(userLogin) == null;
```

Comment:
If the `get` method in `StorageDao` is updated to return an `Optional<User>`, this check should be updated to use `isPresent()` to determine if a user with the given login already exists.